### PR TITLE
Cleanup orphaned code

### PR DESCRIPTION
**Pull request by timroes**
## What

This PR cleans up some orphaned code. It removes files and code not used at all (anymore) as well as removing some exports that were not needed.

I used `ts-prune` to find those orphaned types. This PR does not clean up everything, since I left a couple of exports/types untouched from APIs (that we simply don't use, yet) as well as some types/files I know are currently touched in other PRs. This PR though should cleanup all completely orphaned files.

I removed the `SideView` component, since it's currently not used and didn't have a great accessibility. So if we're planning to have a flyout again in the future, I think we should rebuild that (using headlessui, which we already have in place now), in a better way.

### CODE REVIEW

File: airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx [hunk 1]
```diff
@@ -29,13 +29,6 @@ export const useDocumentationPanelState = () => {
 // @ts-expect-error Default value provided at implementation
 export const documentationPanelContext = createContext<DocumentationPanelContext>();
 
-export const useCloseDocumentationPanelEffect = () => {
-  const { setDocumentationPanelOpen } = useDocumentationPanelState();
-  useEffect(() => {
-    return setDocumentationPanelOpen(false);
-  }, [setDocumentationPanelOpen]);
-};
-
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx:38]: I see this pattern repeated in many places and I wonder if this hook was meant to be used but never happened.

**Review by edmundito — COMMENTED** [review]
Verified all the usages manually and had one observation.

File: airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx (hunk 1 above)
Comment (timroes) [reply airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx:38]: The pattern of closing hooks or the pattern of having code introduced that we never used?

I think once we fully cleaned up dead code we might want to introduce some checks into the CI that at least warn about having unused code around.

**Review by krishnaglick — APPROVED** [review]
`+5 −455` Awesome

File: airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx (hunk 1 above)
Comment (edmundito) [reply airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx:38]: I meant that if you search for `setDocumentationPanelOpen(false)` in the code, you'll see that this snipppet is repeated in multiple places:

```ts
  const { setDocumentationPanelOpen } = useDocumentationPanelState();
  useEffect(() => {
    return setDocumentationPanelOpen(false);
  }, [setDocumentationPanelOpen]);
};
```

Comment (timroes) [reply airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanelContext.tsx:38]: Yeah I agree we could either improve this by making the context methods maybe a `openDocumentationPanel` and `closeDocumentationPanel` or really a hook that auto closes it on unmount. I'd like do that refactoring though not in this PR. But I am fine removing the removal of this code from the PR again, if we want to keep it around for the refactoring to happen.

**Pull request merged**